sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -84,6 +84,26 @@ int pollSocketUntilEventOrTimeout(
   return WSAPoll(&poller, 1, timeout);
 #endif
 }
+
+#ifdef WINDOWS
+// Windows needs this after every write to socket or peformance would be reduced to 1/4 for
+// uploading operation.
+// https://github.com/Azure/azure-sdk-for-cpp/issues/644
+void WinSocketSetBuffSize(curl_socket_t socket)
```

Comment:
Void signatures should be the rarity.  This should return success or failure.

Reply:
There is nothing we want to do if there is a failure on this method.  That's why we just ignore it.
I brought this same approach from how libcurl does it internally for its normal usage (using callbacks),  It is not an error or anything during the transfer if this function is not success